apps/arm/ClawController.cpp

```diff
@@ -2,18 +2,54 @@
 
 ClawController::ClawController(t_actuatorConfig actuatorConfig, 
                        Motor &motor, Encoder &encoder,  
-                       DigitalIn &limSwitchMax, AnalogIn &forceSensor) : 
-        ActuatorController(actuatorConfig, motor, encoder, NULL_DIGITAL_IN, limSwitchMax), r_forceSensor(forceSensor) {};
+                       DigitalIn &limSwitchMax, AnalogIn &forceSensor, Servo &tooltipServo,
+                       float tooltipExtendedAngle_Degrees, float tooltipRetractedAngle_Degrees, float calibrationTimeout_Seconds) : 
+        ActuatorController(actuatorConfig, motor, encoder, NULL_DIGITAL_IN, limSwitchMax), r_forceSensor(forceSensor), r_tooltipServo(tooltipServo),
+        m_tooltipExtendedAngle_Degrees(tooltipExtendedAngle_Degrees), m_tooltipRetractedAngle_Degrees(tooltipRetractedAngle_Degrees),
+        m_calibrationTimeout_Seconds(calibrationTimeout_Seconds) {};
 
+mbed_error_status_t ClawController::setMotorPower_Percentage(float percentage) {
+
+    if (m_mutex.trylock_for(200)) {
+        return ActuatorController::setMotorPower_Percentage(percentage);
+    }
+    else {
+        return MBED_ERROR_MUTEX_LOCK_FAILED;
+    }
+
+    m_mutex.unlock();
+
+    return MBED_SUCCESS;
+}
 
 mbed_error_status_t ClawController::setGapVelocity_CmPerSec(float cmPerSec) {
-    float shaftVelocity_DegreesPerSec = 2.0 * cmPerSec + 0.0 * getGapDistance_Cm(); // TODO find mapping function (velocity dependent on position)
-    return setVelocity_DegreesPerSec(shaftVelocity_DegreesPerSec);
+    
+    if (m_mutex.trylock_for(200)) {
+        float shaftVelocity_DegreesPerSec = convertGapVelocityCmToShaftVelocityDegrees(cmPerSec);
```

Comment:
is this mutex protecting against updating the joint config and updating motion data at same time? should this line be before the trylock to minimize time in lock?

Reply:
for threaded calibration originallt

Reply:
Is it still needed?

Reply:
ya